### ISSUE TITLE
Docs setup

### ISSUE THREAD

**Issue by m10rten** [jetvil/docs#1]
Documentation website could be created with docusaurus.

Make sure to use TS.

Read below for details(docusaurus is not a must).

**Comment by m10rten** [comment]
Docusaurus is ehat I think is easy to setup and future proof once setup correctly.

Please comment under this if you have any other sollution.

Requirements: 

- must use typescript
- must be ready for future features/functions/projects
- must be good and clean looking for simplest UX and DevX when reading docs.
- could have a dark/light theme options.
- could have code block options
- must use markdown or equivalent.

**Comment by adlaonmd** [comment]
I agree with Docusaurus. Looks pretty simple to setup and good for the long run. We should start the initial setup to get things going.

**Comment by m10rten** [comment]
> I agree with Docusaurus. Looks pretty simple to setup and good for the long run. We should start the initial setup to get things going.

Yes, for sure.

When I have the time I will create the basic initial docusaurus setup, nothing more.

Or if anyone else has time to do so, create a PR on the dev branch(if it doesnt exist on time of PR, please notify me again)

**Comment by m10rten** [comment]
Simply created dev.

**Comment by adlaonmd** [comment]
I can do the initial Docusaurus setup.

**Comment by m10rten** [comment]
> I can do the initial Docusaurus setup.

Alright 💪
I will be very critical on the initial setup since its going for a long run.
Dont take my critics as bad, but only as project stability (if I have any feedback at all ✌️😅).

**Comment by m10rten** [comment]
@adlaonmd create a PR and assign me so I get notified.

**Comment by m10rten** [comment]
I just tried out the qwik documentation setup, its under `setup/qwik` in /qwik.

Its solid, but requires a lot of own design I think.
What also gives a lot of config options, which you don't have with docusaurus.

So it made me doubt....

**Comment by m10rten** [comment]
Did the same for Astro in `setup/astro`, not a succes if you ask me.

search engine is OK, but you need to set it up without documentation given.
site looks sloppy and the file are in `.astro` endings...
Rating: not so well.

**Comment by m10rten** [comment]
https://vercel.com/templates/next.js/documentation-starter-kit

Looking into this tuesday when I have time.
seems nice.(nextjs biased a bit)

**Comment by adlaonmd** [comment]
Nextra is not production ready though so it doesn't meet the future proof criteria which is essential.

**Comment by m10rten** [comment]
https://nextjs.org/blog/markdown

Then perhaps we should look into creating our own version.
Perhaps another project: documentation generator.

**Comment by adlaonmd** [comment]
Your call. Whichever seems feasible.

**Comment by m10rten** [comment]
I am a big NextJS fan, making it yourself is also good for config options etc.

Everyone involved gets to speak their mind, so what do you think?

**Comment by adlaonmd** [comment]
I'm a fan of Next.js too so it's not an issue. It's also a great learning opportunity to create another project from scratch, but this time with other people involved.

**Comment by m10rten** [comment]
> I'm a fan of Next.js too so it's not an issue. It's also a great learning opportunity to create another project from scratch, but this time with other people involved.

Alright, I will setup the project and create a new one: jv-doc.

To create a doc website with markdown on top of react websites.
